src/main/java/net/dries007/tfc/common/blocks/IClimateWeatheringBlock.java

```diff
@@ -0,0 +1,194 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.phys.BlockHitResult;
+
+import net.dries007.tfc.common.TFCTags;
+import net.dries007.tfc.common.items.TFCItems;
+import net.dries007.tfc.util.climate.Climate;
+
+public interface IClimateWeatheringBlock
+{
+    float[] AGE_MODIFIERS = {0.5f, 1.0f, 1.0f, 1.0f};
+
+    Block getNext();
+
+    BlockState getNext(BlockState state);
+
+    BlockState getPrevious(BlockState state);
+
+    Block getPrevious();
+
+    default void applyChangeOverTime(BlockState blockState, ServerLevel serverLevel, BlockPos pos, RandomSource random)
+    {
+        if (blockState.getBlock() instanceof IClimateWeatheringBlock climateWeatheringBlock && climateWeatheringBlock.getAge() == TFCWeatherState.FULLY_WEATHERED)
+        {
+            return;
+        }
+
+        // designed to slow weathering after most blocks are exposed
+        Iterator<BlockPos> var8 = BlockPos.withinManhattan(pos, 4, 4, 4).iterator();
+        float count = 0;
+        float overall = 0;
+        while (var8.hasNext())
+        {
+            if (serverLevel.getBlockState(var8.next()).getBlock() instanceof IClimateWeatheringBlock weatheringBlock)
+            {
+                if (weatheringBlock.getAge().ordinal() > this.getAge().ordinal())
+                {
+                    count++;
+                }
+                overall++;
+            }
+        }
+        float neighborModifier = 1;
+        if ((count) / (overall) > 0.4f)
+        {
+            neighborModifier = 0.005f;
+        }
+
+        float randomFloat = random.nextFloat();
+
+        // do climate-affected weathering when blocks are exposed to > 5 blocks of air
+        boolean topExposed = true;
+        for (int y = 1; y < 5; y++)
+        {
+            if (!serverLevel.getBlockState(pos.above(y)).getCollisionShape(serverLevel, pos.above(y)).isEmpty())
+            {
+                topExposed = false;
+                break;
+            }
+        }
+
+        if (topExposed)
+        {
+            float overallChance = getAgeAffectedModifier(this.getAge()) * getClimateAffectedModifier(serverLevel, pos);
+            if (this.getAge() != TFCWeatherState.UNAFFECTED)
+            {
+                overallChance *= neighborModifier;
+            }
+            if (randomFloat < overallChance && randomFloat < getMaterialModifier())
+            {
+                serverLevel.setBlockAndUpdate(pos, getNext(blockState));
+                return;
+            }
+        }
+
+        // do "drip" weathering
+        int adjacentCount = 0;
+        for (Direction direction : Direction.Plane.HORIZONTAL.stream().toList())
```

Comment:
```suggestion
        for (Direction direction : Direction.Plane.HORIZONTAL)
```
Wasteful, does a list copy